### DISCUSSION TITLE
Interactive Quality

### DISCUSSION THREAD

**Discussion by HuyNguyenDePaul** [Kitware/trame#601] [Trame/VTK] [answered]
Hi,

This might be a silly question, but when I was testing out vtkRemoteView:
```
remote_view = vtk.vtkRemoteView(
    view=...,               # Instance of vtkRenderWindow (required)
    ref=...,                # Identifier for this component
    interactive_quality=60, # [0, 100] 0 for fastest render, 100 for best quality
    interactive_ratio=...,  # [0.1, 1] Image size scale factor while interacting
    interactor_events=(     # Enable vtk.js interactor events for method binding
        "events",
        ["EndAnimation"],
    ),
    EndAnimation=end,       # Bind method to the enabled event
)
```

I did not notice any difference when changing the argument passed to the ```interactive_quality``` parameter. I do not work with very complex data or have a very large dataset(s), so this might be my bias. But is there a significant difference between ```interactive_quality = 0``` vs ```interactive_quality = 100```? If so, what type data or geometric shape would showcase this difference?

Best,
Huy Nguyen

**Comment by jourdain** [comment, marked as answer]
This affect the image quality while interacting. In other words, the jpeg compression level. 

You can see the effect of using 0 in the picture below
<img width="1097" alt="Screenshot 2024-10-01 at 08 04 14" src="https://github.com/user-attachments/assets/d8045afb-03f7-48fa-8bd3-9d36232ba582">

**Reply by HuyNguyenDePaul** [reply]
Oh wow, the quality is really bad at 0. That is interesting, because when I was using it for my own application (not cone), I noticed no difference. That is interesting. Thanks for sharing this!

**Reply by jourdain** [reply]
You need to interact with the view to see that... It is usually obvious especially if you go as low as 0. Between 60-80, you get something reasonable.

**Reply by HuyNguyenDePaul** [reply]
Hm that is weird. I might have to take a closer look at my application because I vaguely remember that nothing appeared unusual (rendering time and image quality both) when I set ```interactive_quality = 0```. Albeit I was doing these experiments when I was stressed and tired, I might try it today and I'll get back to you.
